api/config.yaml

```diff
@@ -105,14 +105,25 @@ components:
           type: string
         maxVersion:
           type: string
-    TestConnectionRequest:
+    DataStoreTestConnectionRequest:
       $ref: "#/components/schemas/DataStore"
-    TestConnectionResponse:
+    DataStoreTestConnectionResponse:
       type: object
       properties:
-        successful:
+        allPassed:
           type: boolean
-        errorMessage:
+        checks:
+          type: array
+          items:
+            $ref: "#/components/schemas/DataStoreTestConnectionCheck"
+    DataStoreTestConnectionCheck:
```

Comment:
Depending on the data store and strategy we want to use, the response could have n number of checks